pkg/proxy/proxy.go

```diff
@@ -0,0 +1,202 @@
+package proxy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/brancz/kube-rbac-proxy/pkg/authn"
+	"github.com/brancz/kube-rbac-proxy/pkg/authz"
+	"github.com/golang/glog"
+	yaml "gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+type AuthInterface interface {
```

Comment:
please backport the changes done in https://github.com/brancz/kube-rbac-proxy/pull/14#discussion_r209924958. The interface was removed for, returning plain `(*kubeRBACProxyAuth, error)` instead.